### PR TITLE
Workaround Infinity recompose

### PR DESCRIPTION
**Pull request by VDostoyevskiy**

### CODE REVIEW

File: app/src/main/kotlin/com/hippo/ehviewer/ui/MainActivity.kt [hunk 1]
```diff
@@ -345,7 +346,7 @@ class MainActivity : EhActivity() {
                                                 },
                                                 modifier = Modifier.padding(horizontal = 12.dp),
                                                 icon = {
-                                                    Icon(imageVector = icon, contentDescription = null)
+                                                    Icon(painter = rememberNeverRecomposeVectorPainter(icon), contentDescription = null)
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/MainActivity.kt:349]: Idk if this is a good idea, seems like recomposition is caused by `currentDestination`.

Comment (VDostoyevskiy) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/MainActivity.kt:349]: Nope, root cause is VectorPainter infinitely invalidate itself when recompose, this also happens in galleryscreen

Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/MainActivity.kt:349]: But it shouldn't recompose that often?

Comment (VDostoyevskiy) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/MainActivity.kt:349]: I mean, once rememberImageVector is recomposed
This https://github.com/androidx/androidx/blob/383259a3318443cad02748c73d38423ac2c00405/compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/graphics/vector/VectorPainter.kt#L237 will triggered
And setName invalidate itself, https://github.com/androidx/androidx/blob/383259a3318443cad02748c73d38423ac2c00405/compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/graphics/vector/VectorPainter.kt#L196 invalidateCount++, mutableIntState changed, trigger next recompose
LMAO

Comment (VDostoyevskiy) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/MainActivity.kt:349]: The only thing we can do is prevent rememberImageVector being recomposed

**Review by FooIbar — APPROVED** [review]

**Comment by VDostoyevskiy** [comment]
https://github.com/androidx/androidx/blob/383259a3318443cad02748c73d38423ac2c00405/compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/graphics/vector/Vector.kt#L481
They forget to compare the new name with origin one before invalidate
https://github.com/androidx/androidx/blob/383259a3318443cad02748c73d38423ac2c00405/compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/graphics/vector/VectorPainter.kt#L237
When recompose, they assign the name again, goddamn

**Pull request merged**